src/core/graph.test.js

```diff
@@ -316,7 +316,93 @@ describe("graph", () => {
       });
     });
 
-    describe("in- and out-edges", () => {
+    describe("getInEdges and getOutEdges", () => {
+      describe("type filtering", () => {
+        class ExampleGraphConstructor {
+          graph: Graph<{}, {}>;
+          root: Address;
+          idIncrement: number;
+          inEdges: {[string]: Edge<{}>};
+          outEdges: {[string]: Edge<{}>};
+          constructor() {
+            this.graph = new Graph();
+            this.idIncrement = 0;
+            this.root = this.addNode("ROOT").address;
+            this.inEdges = {
+              A1: this.addEdge("A", "1", true),
+              A2: this.addEdge("A", "2", true),
+              B1: this.addEdge("B", "1", true),
+              B2: this.addEdge("B", "2", true),
+            };
+            this.outEdges = {
+              A1: this.addEdge("A", "1", false),
+              A2: this.addEdge("A", "2", false),
+              B1: this.addEdge("B", "1", false),
+              B2: this.addEdge("B", "2", false),
+            };
+          }
+
+          makeAddress(type: string) {
+            const id = (this.idIncrement++).toString();
+            return {
+              id,
+              type,
+              pluginName: "graph-test",
+              repositoryName: "sourcecred",
+            };
+          }
+
+          addNode(type) {
+            const node = {
+              address: this.makeAddress(type),
+              payload: {},
+            };
+            this.graph.addNode(node);
+            return node;
+          }
+
+          addEdge(nodeType, edgeType, isInEdge) {
+            const node = this.addNode(nodeType);
+            const edge = {
+              address: this.makeAddress(edgeType),
+              src: isInEdge ? node.address : this.root,
+              dst: isInEdge ? this.root : node.address,
+              payload: {},
+            };
+            this.graph.addEdge(edge);
+            return edge;
+          }
+        }
+        ["inEdges", "outEdges"].forEach((choice) => {
```

Comment:
Can get around the string-based dispatch, and avoid the `: any` casts, with something like:
```js
const branches = [{
  choice: "in-edges",
  edges: (egc) => egc.inEdges,
  getEdges: (egc) => (node, options) => egc.graph.getInEdges(node, options);
}, /* the other one */];
branches.forEach(() => { … });
```

Reply:
Done